dask_ml/model_selection/_incremental.py

```diff
@@ -757,7 +859,58 @@ def _get_params(self):
         else:
             return ParameterSampler(self.parameters, self.n_initial_parameters)
 
+    def _stop_on_plateau(self, info, instructions):
+        _best_plateau = {}
+        out = {}
+        for k, steps in instructions.items():
+            records = info[k]
+            current_calls = records[-1]["partial_fit_calls"]
+            if self.max_iter and current_calls >= self.max_iter:
+                out[k] = 0
+            elif self.patience and current_calls >= self.patience:
+                plateau = [
+                    h["score"]
+                    for h in records
+                    if current_calls - h["partial_fit_calls"] <= self.patience
+                ]
+                if all(score <= plateau[0] + self.tol for score in plateau):
```

Comment:
I went with `patience[0]` because it's PyTorch uses `patience[0]` for their `ReduceLROnPlateau` ([source][1]).

Does it matter? I don't think so, but might be wrong (haven't worked an example yet). Both methods are seeing if the most recent scores are within `tol` of each other.

[1]:https://github.com/pytorch/pytorch/blob/294c06538416f5882e9f329695dcae7ff77c218b/torch/optim/lr_scheduler.py#L353-L357

Reply:
> Second note, do we need to treat the "reference" score (plateau[0] or plateau[1]) specially?

Nice catch. Implemented.

> I think that we should be doing either

The first one is implemented now. But they're both the same thing, past some typos in the second one (which is trivially true if the scores are monotonically increasing). Both cases are testing to see if a the scores are in an envelope of "width" `tol` from the "edges" of the envelope. Here's a visualization of what I mean:

<img width="483" alt="screen shot 2018-10-10 at 9 30 08 pm" src="https://user-images.githubusercontent.com/1320475/46777028-9dbcfa00-ccfd-11e8-89bc-6a844a41dc21.png">